### PR TITLE
[Q-SIMPLICITY-JETS-DATA-GO-01] Add Go arithmetic and bytes data jets

### LINKED ISSUE #2085: [Q-SIMPLICITY-JETS-DATA-GO-01] Go arithmetic + bytes jets
contract_version: rubin-linear-issue-contract-v1<br>issue_type: executable_child_slice<br>logical_slice: RUB-490A<br>target_repo: rubin-protocol (clients/go only)<br>parent_tracker: RUB-451 | subfamily_parent: RUB-490<br>design_anchor: RUB-450 v0.4, inherited from RUB-490<br>depends_on: RUB-485<br>execution_policy: EXECUTE_THIS_CHILD_ONLY__DO_NOT_EXPAND_TO_PARENT<br>split_graph_repair: 2026-06-12 staged-go-rust-split

one_invariant: Go implements u64/u128 arithmetic/cmp and bytes_eq/bytes_cmp/bytes_slice jets with deterministic overflow and bounds behavior

architecture_class: staged_single_client_reference<br>system_boundary: implementation<br>accepted_cases: normal operations return expected values/counters<br>rejected_cases: overflow, underflow, zero-length, 32/33 byte boundary, out-of-range slice<br>hostile_cases: integer and byte boundary cases<br>planned_execution_config: GO profile only; implement after live Linear read and full-depth stack pattern recall.<br>mandatory_local_gates: focused Go jet tests plus clients/go go test target

ci_runtime_gate_contract:<br>  github_workflow: parity-gate.yml<br>  staged_split: true<br>  client_slice: go<br>  parity_exempt_required: true<br>  parity_exempt_label: parity-exempt<br>  parity_exemption_pr_body_heading: "### Parity exemption justification"<br>  parity_exemption_reason: "This is a staged single-client child slice. The sibling client and shared evidence are separate Linear issues; do not add sibling-client changes only to satisfy per-PR source lockstep."<br>  fixture_exempt_required: conditional<br>  fixture_exempt_label: fixture-exempt<br>  fixture_exemption_pr_body_heading: "### Fixture exemption justification"<br>  mirror_owner: RUB-554<br>  shared_evidence_owner: RUB-555<br>  non_scope_reinforcement:<br>    - no sibling-client implementation changes<br>    - no shared corpus or fixture changes unless this issue explicitly owns them<br>Scope: Go data jet implementations and focused Go tests

Non-scope: no Rust, no crypto jets, no shared parity corpus, no consensus wiring

done_when: Go data jet tests green for arithmetic and byte boundaries

loc_policy:<br>  target_changed_lines: <=150<br>  hard_cap_changed_lines: <=350<br>  if_exceeds_hard_cap: STOP_AND_SPLIT before push

review_thread_contract:<br>  machine_gate_inputs:<br>    forbidden_surface_tokens:<br>      - same PR Go+Rust lockstep implementation<br>      - sibling-client opportunistic changes<br>      - consensus dispatch wiring unless named in scope<br>      - shared corpus changes unless named in scope

### PR DESCRIPTION
**Pull request by 2tbmz9y2xt-lang**
Invariant:
Go data jet helpers implement checked u64/u128 arithmetic, ordering, and bytes equality/comparison/slicing with deterministic overflow and bounds behavior.

Layer:
implementation / tests

Files changed:
- `clients/go/consensus/simplicity/data_jets.go`
- `clients/go/consensus/simplicity/data_jets_test.go`

Tests:
- `scripts/dev-env.sh -- bash -lc 'cd clients/go && go test ./consensus/simplicity'` - PASS
- `scripts/dev-env.sh -- bash -lc 'cd clients/go && go test ./...'` - PASS
- `scripts/dev-env.sh -- bash -lc 'cd clients/go && go vet ./...'` - PASS
- `git diff --check` - PASS
- `check_complexity.py --repo . --base-ref origin/main clients/go/consensus/simplicity/data_jets.go clients/go/consensus/simplicity/data_jets_test.go` - PASS, added CCN 20
- `rubin-stack-codacy-coverage.sh . origin/main` - PASS, diff coverage 100.00%, variation +0.02%
- `rubin-precommit-one-review --repo . --base-ref origin/main --include-base-diff` - PASS
- isolated stock code-review - No findings
- `rubin-push --coverage-cmd 'rubin-stack-codacy-coverage.sh . origin/main'` - PASS

Behavior impact:
Adds pure Go helper functions and focused tests for data jets. This PR does not wire the helpers into Simplicity dispatch.

Compatibility impact:
No wire format, fixture, Rust, or shared corpus change.

Known non-goals:
- No Rust mirror.
- No shared parity corpus.
- No crypto jets.
- No consensus dispatch wiring.
- No generated artifact changes.

### Parity exemption justification

RUB-553 is a Go-only child slice. The sibling Rust mirror and cross-client parity work are explicitly outside this PR. The diff adds Go helper implementations and Go package tests only; it does not change shared conformance fixtures, generated corpora, dispatch wiring, or Rust code. This block is the parity-exempt justification required by the parity-smoke gate.

Closes #2085

### CODE REVIEW

File: clients/go/consensus/simplicity/data_jets_test.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	}
+	if got := EvaluateBytesSliceJet(src, 5, 2); got.Accepted || got.Cost != 2 {
+		t.Fatalf("bytes_slice out of range = %+v, want rejected cost=2", got)
+	}
+	if got := EvaluateBytesSliceJet(src, ^uint64(0), 1); got.Accepted || got.Cost != 2 {
+		t.Fatalf("bytes_slice overflow = %+v, want rejected cost=2", got)
+	}
+	if got := EvaluateBytesSliceJet(nil, 0, ^uint64(0)); got.Accepted || got.Cost != 576460752303423489 {
+		t.Fatalf("bytes_slice max len = %+v, want rejected cost=576460752303423489", got)
+	}
```
[the remark this answers is not in the document]
Comment (2tbmz9y2xt-lang) [reply clients/go/consensus/simplicity/data_jets_test.go]: Fixed in 236374a: the max bytes_slice expected cost now derives from dataJetFlatCost and bytesJetChunkLen without calling bytesJetCost, and the failure message prints the computed expected value.
